shared/templates/create_mount_options.py

```diff
@@ -8,44 +8,143 @@
+    oval="./oval/mount_option_{0}.xml",
+)
+
+
+class Skipped(Exception):
+    pass
+
+
+class MountOptionTarget(object):
+    TEMPLATE_FILE_BASE = None
+    OUTPUT_FORMAT_STRING = None
+
+    def __init__(self, generator, target):
+        self.OUTPUT_FORMAT_STRING = OUTPUTS_FORMAT_STRINGS[target]
+        self.generator = generator
+        self.TEMPLATE_FILE_BASE = self.TEMPLATE_FILE_BASE.format(target=target.upper())
+        self.template_file = self.TEMPLATE_FILE_BASE
+
+        self._mount_point = ""
+        self._mount_option = ""
+        self._point_id = ""
+        self._output_id_template = "{point_id}_{mount_option}"
+        self._assert_mount_exists = True
+        self._output_fname = None
+
+    def process(self, mount_point, mount_option, assert_mount_exists):
+        point_id = re.sub('[-\./]', '_', mount_point).lstrip("_")
+
+        try:
+            self._set_correct_values(mount_point, mount_option, point_id)
+        except Skipped:
+            return
+
+        self._output_fname = self.OUTPUT_FORMAT_STRING.format(
+            self._output_id_template.format(
+                point_id=self._point_id, mount_option=self._mount_option)
+        )
+        self._assert_mount_exists = assert_mount_exists
+        self._process()
+
+    def _process(self):
+        raise NotImplementedError("You are supposed to use a derived class.")
+
+    def _set_correct_values(self, mount_point, mount_option, point_id):
+        self._mount_point = mount_point
+        self._point_id = point_id
+        self._mount_option = mount_option
+
+
+class RemediationTarget(MountOptionTarget):
+    TEMPLATE_FILE_BASE = "./template_{target}_mount_option"
+
+    def _process(self):
+        mount_has_to_exist = "yes" if self._assert_mount_exists else "no"
+        self.generator.file_from_template(
+            self.template_file,
+            {
+                "%MOUNT_HAS_TO_EXIST%": mount_has_to_exist,
+                "%MOUNTPOINT%": self._mount_point,
+                "%MOUNTOPTION%": re.sub(' ', ',', self._mount_option),
+            },
+            self._output_fname,
+            ""
+        )
+
+    def _set_correct_values(self, mount_point, mount_option, point_id):
+        super(RemediationTarget, self)._set_correct_values(mount_point, mount_option, point_id)
+        if mount_point.startswith("var_"):
+            self._point_id = re.sub(r"^var_(.*)", r"\1s", mount_point)
+            self.template_file = "{0}_var".format(self.TEMPLATE_FILE_BASE)
+            self._output_id_template = "{mount_option}_{point_id}"
+        elif not mount_point.startswith("/"):  # no path, but not a variable either
+            raise Skipped("No template available yet")
+
+
+class OvalTarget(MountOptionTarget):
+    TEMPLATE_FILE_BASE = "./template_OVAL_mount_option"
+
+    def __init__(self, generator):
+        super(OvalTarget, self).__init__(
+            generator, "oval")
+
+    def _set_correct_values(self, mount_point, mount_option, point_id):
```

Comment:
I have renamed it, but anyone please help me if you have a better name idea. I think that the method does one thing, so it justified, but finding a not-too-long name for it may be a challenge.